File: lc_go/lcGo/slidingWindows/countSubarrays2444.go
package main

import "fmt"

func countSubarrays(nums []int,minK,maxK int)(ans int64){
	n := len(nums)

	left,minKPos,maxKPos := -1,-1,-1

	for i := range nums[:n] {
		if nums[i] < minK || nums[i] > maxK {
			left = i
		}
		if nums[i] == minK {
			minKPos = i
		}
		if nums[i] == maxK {
			maxKPos = i 
		}
		validStart := min(minKPos,maxKPos)
		if validStart > left {
			ans += int64(validStart - left)
		}

	}
	return 
}

func main(){
	nums := []int{1,3,5,2,7,5}
	minK,maxK := 1,5
	res := countSubarrays(nums,minK,maxK)
	fmt.Println(res)
}

File: lc_py/lcPy/sliding_window/CountSubarrays2444.py
class CountSubarrays:
    def countSubarrays(self,nums: List[int],minK: int,maxK: int) -> int:
        ans = 0
        n = len(nums)
        minK_pos = maxK_pos = left = -1

        for i in range(n):
            if nums[i] < minK or nums[i] > maxK:
                left = i
            if nums[i] == minK:
                minK_pos = i
            if nums[i] == maxK:
                maxK_pos = i
            valid_start = min(minK_pos,maxK_pos)
            if valid_start > left:
                ans += valid_start - left
        return ans